activators/caila/src/main/kotlin/com/justai/jaicf/activator/caila/CailaThresoldProcessor.kt

```diff
@@ -0,0 +1,35 @@
+package com.justai.jaicf.activator.caila
+
+import com.justai.jaicf.activator.caila.dto.CailaInferenceResultData
+import kotlin.math.max
+
+class CailaThresholdProcessor(
+    settings: CailaNLUSettings
+) {
+    private val patternThreshold = settings.intentThresholds.patterns ?: settings.confidenceThreshold
+    private val phrasesThreshold = settings.intentThresholds.phrases ?: settings.confidenceThreshold
+
+    //  Due to different thresholds for patters and phrases result.confidence can be changed after this method
```

Comment:
It's better to write comments before method in javadoc format, not regular comments